test/e2e_node/testdeviceplugin/device-plugin.go

```diff
@@ -0,0 +1,237 @@
+			if dp.errorInjector != nil {
+				if err := dp.errorInjector("ListAndWatch"); err != nil {
+					return err
+				}
+			}
+		}
+	}
+}
+
+func (dp *DevicePlugin) Allocate(ctx context.Context, request *kubeletdevicepluginv1beta1.AllocateRequest) (*kubeletdevicepluginv1beta1.AllocateResponse, error) {
+	result := new(kubeletdevicepluginv1beta1.AllocateResponse)
+
+	dp.callsSync.Lock()
+	dp.calls = append(dp.calls, "Allocate")
+	dp.callsSync.Unlock()
+
+	for _, r := range request.ContainerRequests {
+		response := &kubeletdevicepluginv1beta1.ContainerAllocateResponse{}
+		for _, id := range r.DevicesIDs {
+			fpath, err := os.CreateTemp("/tmp", fmt.Sprintf("%s-%s", dp.uniqueName, id))
+			gomega.Expect(err).To(gomega.Succeed())
+
+			response.Mounts = append(response.Mounts, &kubeletdevicepluginv1beta1.Mount{
+				ContainerPath: fpath.Name(),
+				HostPath:      fpath.Name(),
+			})
+		}
+		result.ContainerResponses = append(result.ContainerResponses, response)
+	}
+
+	return result, nil
+}
+
+func (dp *DevicePlugin) PreStartContainer(ctx context.Context, request *kubeletdevicepluginv1beta1.PreStartContainerRequest) (*kubeletdevicepluginv1beta1.PreStartContainerResponse, error) {
+	return nil, nil
+}
+
+func (dp *DevicePlugin) GetPreferredAllocation(ctx context.Context, request *kubeletdevicepluginv1beta1.PreferredAllocationRequest) (*kubeletdevicepluginv1beta1.PreferredAllocationResponse, error) {
+	return nil, nil
+}
+
+func (dp *DevicePlugin) RegisterDevicePlugin(uniqueName, resourceName string, devices []kubeletdevicepluginv1beta1.Device) error {
+	dp.devicesSync.Lock()
+	dp.devices = devices
+	dp.devicesSync.Unlock()
+
+	devicePluginEndpoint := fmt.Sprintf("%s-%s.sock", "test-device-plugin", uniqueName)
+	dp.uniqueName = uniqueName
+
+	// Implement the logic to register the device plugin with the kubelet
+	// Create a new gRPC server
+	dp.server = grpc.NewServer()
+	// Register the device plugin with the server
+	kubeletdevicepluginv1beta1.RegisterDevicePluginServer(dp.server, dp)
+	// Create a listener on a specific port
+	lis, err := net.Listen("unix", kubeletdevicepluginv1beta1.DevicePluginPath+devicePluginEndpoint)
+	if err != nil {
+		return err
+	}
+	// Start the gRPC server
+	go func() {
+		err := dp.server.Serve(lis)
+		gomega.Expect(err).To(gomega.Succeed())
+	}()
+
+	// Create a connection to the kubelet
+	conn, err := grpc.Dial(kubeletdevicepluginv1beta1.KubeletSocket,
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithContextDialer(func(ctx context.Context, addr string) (net.Conn, error) {
+			return net.Dial("unix", addr)
+		}),
+	)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err := conn.Close()
+		gomega.Expect(err).To(gomega.Succeed())
+	}()
+
+	// Create a client for the kubelet
+	client := kubeletdevicepluginv1beta1.NewRegistrationClient(conn)
+
+	// Register the device plugin with the kubelet
+	_, err = client.Register(context.Background(), &kubeletdevicepluginv1beta1.RegisterRequest{
```

Comment:
Would it make sense to pass test content to this function and use it here instead of creating a new context?
The same applies to the `getNodeResourceValues`.